lib/Model/Query.js

```diff
@@ -158,7 +158,7 @@ function Query(model, collectionName, expression, options) {
   // because otherwise maybeUnload would be looping through the entire results
   // set of each query on the same collection for every doc checked
   //
-  // Map of id -> true
+  // Map of id -> count of ids
   this.idMap = {};
```

Comment:
True. The problem would result in a doc not getting locally cleaned up, so honestly, the impact isn't likely to be a concern. I'd probably not worry about it for now.